mathics/builtin/arithmetic.py

```diff
@@ -2189,15 +2195,18 @@ class Piecewise(SympyFunction):
 
     def apply(self, items, evaluation):
         "%(name)s[items__]"
-        result = self.to_sympy(Expression("Piecewise", *items.get_sequence()))
+        result = self.to_sympy(
+            Expression("Piecewise", *items.get_sequence()), evaluation=evaluation
```

Comment:
Great! This is what I was expecting to see!

Reply:
> > I think this is ready now: the function `predicate_evaluation(predicate, evaluation)` looks at $Assumptions and implement some replacements on predicate accordingly. This function is called then in the evaluation of `Piecewise`, and `ConditionalExpression`, in a way that propagates automatically to `Integrate`.
> 
> A superficial look through and I think this is exactly the right thing to do. However I want to pull this down and try it and maybe even experiment with the code to understand what's up with sympy mentioned in one of the comments. And probably the weekend is when I'll have an opportunity to do.

Update: I realized that in WMA,  logic expressions are evaluated using assumptions just if you surround them by  ``Simplify``.  On the other hand, `Integrate` uses the option  `Assumptions` (default `$Assumptions`) before return the result. Now this behavior is implemented.

Reply:
Also, I found -I guess- a safe way to reproduce the use of `ConditionalExpression` instead of ``Piecewise`` in the return of Integrate.